astropy/utils/metadata.py

```diff
@@ -81,7 +176,18 @@ def merge(left, right, merge_func=concat, metadata_conflicts='warn'):
 
         else:
             try:
-                out[key] = merge_func(left[key], right[key])
+                if merge_func is None:
+                    for left_type, right_type, merge_cls in MERGE_STRATEGY_CLASSES:
+                        if not merge_cls.enabled:
```

Comment:
Yes, it was funny to see that you basically described how `add_enabled_units` works in your one-but-last comment!